### ISSUE TITLE
Running npm-cache-list locks up

### ISSUE THREAD

**Issue by supernova2468** [yilmazdurmaz/npm-cache-list#1]
When I run the command the console doesn't do anything it just sits there blinking. 

After trying to create this on my own I believe it is this line that hangs

``child_process.execSync("npm config get cache").toString().trim()``

**Comment by yilmazdurmaz** [comment]
Hi @supernova2468,
This one is my first tool ever and thank you for trying it.

In theory, there is only 1 possible part it may hang and it is getting the list of cached packages and searching the list, and this may happen only due to large numbers of them. Listing all packages starts almost immediately and takes time to write list to console, while searching first takes the list and searches for the terms

``npm-cache-list all``
> Cache is at 't:\Tools\node-v11.12.0-win-x64\npm-cache'
> Listing all versioned packages, this will take some time

``npm-cache-list list express``
> Cache is at 't:\Tools\node-v11.12.0-win-x64\npm-cache'
> Listing packages including search terms: [ 'express' ]

If you don't see any one of the above 2 examples, then node itself might have the issue to start the tool. For now, as I don't know what system/environment/versions you use, I can only suggest a simple test to use (and a personal fix depending the result). You have already done some tests but please follow these steps and report back if possible.

1. Just to be sure, open a terminal and just run ``npm-cache-list``. If it runs and shows help, we can dive into code. Otherwise we can say there is a problem in your system to prevent the tool running.
2. Find ``npm-cache-list`` folder in your local or global ``node_modules`` folder (global path: ``npm config get prefix``). Open ``index.js`` and add ``console.log("Trying to get cache path");`` at line 9, then run the tool again. If you see this message before the tool locks-up we can safely assume the problem is at line 10.
3. Open a console and start node interactive by just running ``node`` command, then copy-paste and run this code: ``child_process.execSync("npm config get cache").toString().trim()``. If it fails we can say there is a problem with your version. If runs, there is a problem running that code through a script.
4. Open a new JavaScript file (ie. ``one-line-test.js``), put the same code line in it, save and run. (``node one-line-test.js``)
Report back your findings when you have time, and
5. While trying to process the results, comment out lines 10 and 11, and try to use the tool again. It should still work and give you a list of packages.

**Comment by yilmazdurmaz** [comment]
Thanks for the feedback,
It seems there is a problem with child_process on LTS (10.x) Version. I was developing with 11.x version, and being new in the field I was not aware of such problems.
I will later check about it and might remove those 2 lines later. Meanwhile, you can use the tool without those 2 lines as their sole purpose is just to show where the cache is located of which you can get running the config command in terminal.

**Comment by yilmazdurmaz** [comment]
@supernova2468, may I ask for another test on your PC, if possible?

I have only 3 64bit versions installed on a Win-10, one of them being node 10.16.0. I have just tested on it, and the tool works without any problem. I am now a bit unsure if it is about being 32bit (again another unknown about your system) or new node version has a problem.

Unfortunately, I have limited network for now (phone data) and cannot cover other versions right now. Is it possible for you to try another 10.x release (10.16.0 preferred) from "https://nodejs.org/en/download/releases/"? Because, if not the node, then there is this possibility your windows installation or settings has something to prevent ``child_process`` to run.
